homeassistant/components/camera/yi.py

```diff
@@ -57,68 +55,67 @@ def __init__(self, hass, config):
-
-    @property
-    def name(self):
-        """Return the name of this camera."""
-        return self._name
+        self._name = config[CONF_NAME]
+        self.host = config[CONF_HOST]
+        self.port = config[CONF_PORT]
+        self.path = config[CONF_PATH]
+        self.user = config[CONF_USERNAME]
+        self.passwd = config[CONF_PASSWORD]
 
     @property
     def brand(self):
         """Camera brand."""
         return DEFAULT_BRAND
 
-    def get_latest_video_url(self):
-        """Retrieve the latest video file from the customized Yi FTP server."""
-        from ftplib import FTP, error_perm
+    @property
+    def name(self):
+        """Return the name of this camera."""
+        return self._name
 
-        ftp = FTP(self.host)
-        try:
-            ftp.login(self.user, self.passwd)
-        except error_perm as exc:
-            _LOGGER.error('There was an error while logging into the camera')
-            _LOGGER.debug(exc)
-            return False
+    async def _get_latest_video_url(self):
+        """Retrieve the latest video file from the customized Yi FTP server."""
+        from aioftp import Client, StatusCodeError
 
+        ftp = Client()
         try:
-            ftp.cwd(self.path)
-        except error_perm as exc:
-            _LOGGER.error('Unable to find path: %s', self.path)
-            _LOGGER.debug(exc)
-            return False
-
-        dirs = [d for d in ftp.nlst() if '.' not in d]
-        if not dirs:
-            _LOGGER.warning("There don't appear to be any uploaded videos")
-            return False
-
-        latest_dir = dirs[-1]
-        ftp.cwd(latest_dir)
-        videos = ftp.nlst()
-        if not videos:
-            _LOGGER.info('Video folder "%s" is empty; delaying', latest_dir)
-            return False
-
-        return 'ftp://{0}:{1}@{2}:{3}{4}/{5}/{6}'.format(
-            self.user, self.passwd, self.host, self.port, self.path,
-            latest_dir, videos[-1])
+            await ftp.connect(self.host)
+            await ftp.login(self.user, self.passwd)
+
+            await ftp.change_directory(self.path)
+            dirs = []
+            for path, attrs in await ftp.list():
+                if attrs['type'] == 'dir' and '.' not in str(path):
+                    dirs.append(path)
+            latest_dir = dirs[-1]
+            await ftp.change_directory(latest_dir)
+
+            videos = []
+            for path, _ in await ftp.list():
+                videos.append(path)
+            if not videos:
+                _LOGGER.info('Video folder "%s" empty; delaying', latest_dir)
+                return False
+
+            return 'ftp://{0}:{1}@{2}:{3}{4}/{5}/{6}'.format(
+                self.user, self.passwd, self.host, self.port, self.path,
+                latest_dir, videos[-1])
+        except (ConnectionRefusedError, StatusCodeError) as err:
+            print('There was an issue: {0}'.format(err))
+            return
```

Comment:
Inconsistent return - once you're returning `False`, once a string and here `None`. It does work but I would suggest switching both error cases to `return None`.